assignment3.txt

```diff
@@ -0,0 +1,26 @@
+Emma Ladouceur
+
+Assignment Answers
+
+
+1. I dont' think it is valid.
+	First after the %d what is the code printing? That means it should be preparing for an integer and there is nothing defined nor is there a variable name following in. 
+	Secondly, there is a semicolon before the i.
+
+
+2. 
+There should not be a semi colon after the first set of () because it will go straight to the {}.
+
+
+3. I'm still a little confused about this, but I'm going to try it out.
+
+i. There are no factorials or squares or cubes or logs so I would say this is just 0(1) in terms of complexity.
+
+ii.
+Okay. This one is a little more complicated I think because of the nested loops and the if statement but I don't really understand how it moves up so I would just guess that it jumps up to O(logn)
```

Comment:
This one is tough! The answer is actually O(n^4) - again, for loops that run from 0 to n have the time complexity O(n). So the top loop is O(n), the middle loop is O(n^2) iterations because it iterates from 1 to i^2, which is equivalent to n^2. The if statement is O(1 / n) times because of the modulo function. Lastly, the inner loop  is O(n^2) iterations because it iterates from 0 to j and j = i^2 = n^2.

If you're still confused after Ms. Hsu explains, ask for clarification.  

Therefore, the sum = -(1/12)*n + (3/8)*n^2 - (5/12)*n^3 + (1/8)*n^4  = O(n^4)